lib/granite/action/precondition.rb

```diff
@@ -0,0 +1,38 @@
+module Granite
+  class Action
+    class Precondition < BasicObject
+      UNDEFINED = ::Object.new.freeze
+
+      def self.description(text = UNDEFINED)
+        case text
+        when UNDEFINED
+          @description
+        else
+          @description = text
+        end
+      end
+
+      def initialize(context)
+        @context = context
+      end
+
+      def call(*)
+        fail NotImplementedError, "#call method must be implemented for #{self.class}"
+      end
+
+      def method_missing(method_name, *args, &blk)
```

Comment:
`method_missing(method_name, *args, **kwargs, &blk)` for ruby-3 compat.

Reply:
This does not work with ruby 2.6.